packages/react/src/components/tree-view/tree-view.stories.tsx

```diff
@@ -26,14 +27,40 @@ class TreeViewDemo extends React.Component<TreeViewDemoProps, TreeViewDemoState>
   }
 
   render() {
-    return treeViewTemplate({
-      collection: this.state.collection,
-      onOpenItem: e => {
-        return this.props.onOpenItem(e.detail, collection => this.setState({ ...this.state, collection }));
-      },
-      onCloseItem: e => this.props.onCloseItem(e.detail, collection => this.setState({ ...this.state, collection })),
-      onClickItem: e => this.props.onClickItem(e.detail)
-    });
+    return (
+      <div className="container" style={{ display: 'flex', width: '100%' }}>
+        <div style={{ flex: 1 }}>
+          {treeViewTemplate({
+            collection: this.state.collection,
+            onOpenItem: e => {
+              return this.props.onOpenItem(e.detail, collection => this.setState({ ...this.state, collection }));
+            },
+            onCloseItem: e => this.props.onCloseItem(e.detail, collection => this.setState({ ...this.state, collection })),
+            onClickItem: e => this.props.onClickItem(e.detail.path, e.detail.originalEvent, collection => this.setState({ ...this.state, collection }))
+          })}
+        </div>
+        <div style={{ flex: 1 }}>
+          <label htmlFor="treeViewSearchInput">Zoek</label>
+          <input
+            type="text"
+            id="treeViewSearchInput"
+            onChange={e => {
+              const count: number = this.props.onFilter(e.target.value, collection => this.setState({ ...this.state, collection }))
+              const target = document.getElementById('treeViewSearchResultCount');
+              if (target) {
+                target.innerText = `${(count > 0 ? count : 'geen')} resulta${(count === 1 ? 'at' : 'ten')} gevonden ${(e.target.value ? ` voor "${e.target.value}"` : '')}`;
```

Comment:
De `target.innerText` kan je op regel 48 bij de `this.setState()` opnemen in de state van dit React component. Daarna kan je op regel 55 de tekst vanuit de state in de `span` zetten.

Dan werkt dit eigenlijk net zoals op regel 59: De state is de bron en je zoekt geen HTML element op maar blijft binnen de React/JSX wereld.

Je zou kunnen overwegen dat deze "search result count" tekst een argument is van de callback function van `this.props.onFilter`: dan is de logica voor beide demo componenten (lit en React) gecentraliseerd.